File: src/main/java/com/hedvig/backoffice/services/tickets/dto/TicketRevisionDto.java
package com.hedvig.backoffice.services.tickets.dto;

import com.hedvig.backoffice.services.tickets.TicketChangeType;
import lombok.Value;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalTime;

@Value
public class TicketRevisionDto {
  String assignedTo;
  Float manualPriority;
  LocalDate remindDate;
  LocalTime remindTime;
  String remindMessage;
  TicketStatus status;
  Instant changedAt;
  TicketChangeType changeType;
  String changedBy;
  String description;
}

File: src/main/java/com/hedvig/backoffice/services/tickets/dto/TicketRevisionDto.kt
package com.hedvig.backoffice.services.tickets.dto

import java.time.Instant
import java.time.LocalDate
import java.time.LocalTime

data class TicketRevisionDto(
  val assignedTo: String? = null,
  val manualPriority: Float? = null,
  val remindDate: LocalDate? = null,
  val remindTime: LocalTime? = null,
  val remindMessage: String? = null,
  val status: TicketStatus,
  val changedAt: Instant,
  val changeType: TicketChangeType,
  val changedBy: String,
  val description: String? = null
)
